engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlanConfiguration.java

```diff
@@ -715,8 +1438,8 @@ public StramElement getElement()
     elementMaps.put(StramElement.OUTPUT_PORT, PortConf.class);
   }
 
-  private Conf getConf(StramElement element, Conf ancestorConf) {
-    if (element == ancestorConf.getElement()) {
+  private static Conf getConf(StramElement element, Conf ancestorConf) {
```

Comment:
Sorry I lied. This method does an implicit check for correctness, and throws a NPE if something is invalid. Very confusing, I'll throw an explicit error and add the documentation.